File: src/main/java/io/github/emcw/adapters/DurationAdapter.java
package io.github.emcw.adapters;

import com.google.gson.*;

import java.lang.reflect.Type;
import java.time.Duration;

public class DurationAdapter implements JsonSerializer<Duration>, JsonDeserializer<Duration> {
    @Override
    public JsonElement serialize(Duration duration, Type type, JsonSerializationContext ctx) {
        return new JsonPrimitive(duration.toString());
    }

    @Override
    public Duration deserialize(JsonElement el, Type type, JsonDeserializationContext ctx) throws JsonParseException {
        return Duration.parse(el.getAsString());
    }
}

File: src/main/kotlin/io/github/emcw/adapters/DurationAdapter.kt
package io.github.emcw.adapters

import com.google.gson.*
import java.lang.reflect.Type
import java.time.Duration

class DurationAdapter : JsonSerializer<Duration>, JsonDeserializer<Duration> {
    override fun serialize(duration: Duration, type: Type, ctx: JsonSerializationContext): JsonElement {
        return JsonPrimitive(duration.toString())
    }

    @Throws(JsonParseException::class)
    override fun deserialize(el: JsonElement, type: Type, ctx: JsonDeserializationContext): Duration {
        return Duration.parse(el.asString)
    }
}